login_session.go

```diff
@@ -109,9 +111,16 @@ func (s LoginSession) Clear() error {
 }
 
 // Frontpage returns the submissions on the logged-in user's personal frontpage.
-func (s LoginSession) Frontpage() ([]*Submission, error) {
+func (s LoginSession) Frontpage(sort popularitySort, params ListingOptions) ([]*Submission, error) {
+	v, err := query.Values(params)
+	if err != nil {
+		return nil, err
+	}
+
+	baseUrl, err := url.Parse(fmt.Sprintf("http://www.reddit.com/%s/.json?%s", sort, v.Encode()))
```

Comment:
There's no point in using `url.Parse` especially not if you aren't even checking the error.